WalletWasabi.Gui/Global.cs

```diff
@@ -518,21 +520,29 @@ public async Task InitializeWalletServiceAsync(KeyManager keyManager)
 					await Task.Delay(100, token);
 				}
 
-				WalletService = new WalletService(BitcoinStore, keyManager, Synchronizer, Nodes, DataDir, Config.ServiceConfiguration, FeeProviders, BitcoinCoreNode);
+				walletService = new WalletService(BitcoinStore, keyManager, Synchronizer, Nodes, DataDir, Config.ServiceConfiguration, FeeProviders, BitcoinCoreNode);
 
 				Logger.LogInfo($"Starting {nameof(WalletService)}...");
-				await WalletService.StartAsync(token);
+				await walletService.StartAsync(token);
 				Logger.LogInfo($"{nameof(WalletService)} started.");
 
 				token.ThrowIfCancellationRequested();
 
-				TransactionBroadcaster.AddWalletService(WalletService);
-				CoinJoinProcessor.AddWalletService(WalletService);
+				TransactionBroadcaster.AddWalletService(walletService);
+				CoinJoinProcessor.AddWalletService(walletService);
+
+				walletService.TransactionProcessor.WalletRelevantTransactionProcessed += TransactionProcessor_WalletRelevantTransactionProcessed;
+				walletService.ChaumianClient.OnDequeue += ChaumianClient_OnDequeued;
 
-				WalletService.TransactionProcessor.WalletRelevantTransactionProcessed += TransactionProcessor_WalletRelevantTransactionProcessed;
-				WalletService.ChaumianClient.OnDequeue += ChaumianClient_OnDequeued;
+				if(WalletService is  null)
+				{
+					WalletService = walletService;
```

Comment:
Strange. Calling create will set an internal prop in the class? Maybe it is just the naming.

Reply:
Thats a temporary hack, because really I should get rid of WalletService property, but then this PR turns into a 300 - 400 liner.

Reply:
Thats a temporary hack, because really I should get rid of WalletService property, but then this PR turns into a 300 - 400 liner.

Reply:
Mostly we need to keep this property filled because of Rpc code, and some other changes for multiwallet, that are complex and need reviewing seperately, like cleanup when app closes, and status messages

Reply:
https://github.com/zkSNACKs/WalletWasabi/blob/a3670f2412bc11212ebd528f23b3e4549c7ba97a/WalletWasabi.Gui/Wallet.cs#L50

this is the direction it is going towards.